suite-common/wallet-core/src/device/deviceReducer.ts

```diff
@@ -115,17 +116,29 @@ const connectDevice = (draft: State, device: Device) => {
     // fill draft with not affected devices
     otherDevices.forEach(d => draft.devices.push(d));
 
-    // prepare new device
+    const deviceInstance = features.passphrase_protection
+        ? deviceUtils.getNewInstanceNumber(draft.devices, device) || 1
+        : undefined;
+
+    const isMobile = getEnvironment() === 'mobile';
+
+    let useEmptyPassphrase: boolean;
+    if (isMobile && typeof deviceInstance === 'number' && deviceInstance === 1) {
```

Comment:
typeof number condition is redundant?
```suggestion
    if (isMobile && deviceInstance === 1) {
```

Reply:
There is fallback to undefined, so I'm keeping typescript happy here and didn't want to tweak the instance logic in case there is some parts of code that rely on that undefined.
![image](https://github.com/trezor/trezor-suite/assets/36101761/ae5bac8c-ac93-4cd1-b4cf-d23b8f4bcc92)

Reply:
https://github.com/trezor/trezor-suite/pull/12036/commits/30c1cb964118696dc7da0d97790f1cc2ee0eabe1

Reply:
I would be less confused by `typeof deviceInstance !== 'undefined'` but nevermind...